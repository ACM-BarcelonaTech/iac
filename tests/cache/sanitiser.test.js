const sanitiseGlobalConfiguration = require('../../src/cache/sanitiser');

jest.mock('../../src/config');

describe('sanitiser', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  describe('sanitise global configuration', () => {
    let configuration;

    afterEach(() => expect(configuration).toMatchSnapshot());

    it('sanitise if no config provided, dev env', () => {
      configuration = sanitiseGlobalConfiguration({});
    });

    it('sanitise to default retry delay if delay is smaller then default', () => {
      configuration = sanitiseGlobalConfiguration({ retryDelay: 1 });
    });

    it('sanitise to new retry delay if delay is bigger then default', () => {
      configuration = sanitiseGlobalConfiguration({ retryDelay: 20000 });
    });

    it('sanitise with overridden skipAvailabilityZoneCheck', () => {
      configuration = sanitiseGlobalConfiguration({ skipAvailabilityZoneCheck: true });
    });

    it('sanitises the l1CacheSettings using the default values, when it\'s empty', () => {
      configuration = sanitiseGlobalConfiguration({ l1CacheSettings: {} });
    });

    it('sanitises the l1CacheSettings using the default values, when it\'s invalid', () => {
      configuration = sanitiseGlobalConfiguration({
        l1CacheSettings: {
          size: 'invalid',
          ttl: 'invalid',
          minLatencyToStore: 'invalid',
        },
      });
    });

    it('sanitises the l1CacheSettings - uses the provided config when valid', () => {
      configuration = sanitiseGlobalConfiguration({
        l1CacheSettings: {
          size: '2000',
          ttl: 120,
          minLatencyToStore: 10,
        },
      });
    });

    it('sanitises the l1CacheSettings - uses the provided config when valid (minLatencyToStore=0)', () => {
      configuration = sanitiseGlobalConfiguration({
        l1CacheSettings: {
          size: '2000',
          ttl: 120,
          minLatencyToStore: 0,
        },
      });
    });
  });
});
