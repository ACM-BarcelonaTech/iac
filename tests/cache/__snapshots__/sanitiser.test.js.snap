// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sanitiser sanitise global configuration sanitise if no config provided, dev env 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitise to default retry delay if delay is smaller then default 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitise to new retry delay if delay is bigger then default 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 20000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitise with overridden skipAvailabilityZoneCheck 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": true,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitises the l1CacheSettings - uses the provided config when valid (minLatencyToStore=0) 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitises the l1CacheSettings - uses the provided config when valid 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitises the l1CacheSettings using the default values, when it's empty 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;

exports[`sanitiser sanitise global configuration sanitises the l1CacheSettings using the default values, when it's invalid 1`] = `
Object {
  "cacheDuration": 60,
  "l1CacheSettings": Object {
    "minLatencyToStore": 50,
    "size": 1000,
    "ttl": 60000,
  },
  "master": Object {
    "az": "eu-west-2b",
    "host": "biomage-cache-001.lmoszc.0001.euw2.cache.amazonaws.com",
    "port": "6379",
  },
  "redisGetTimeout": 3000,
  "retryDelay": 10000,
  "skipAvailabilityZoneCheck": false,
  "slaves": Array [
    Object {
      "az": "eu-west-2b",
      "host": "biomage-cache-002.lmoszc.0001.euw2.cache.amazonaws.com",
      "port": "6379",
    },
  ],
}
`;
